From 18030fd8e0959d2a7bf742c2ea3184ea40c16981 Mon Sep 17 00:00:00 2001
From: Mandy Lavi <mandy.lavi@freescale.com>
Date: Mon, 2 Jun 2014 21:57:27 +0300
Subject: [PATCH 719/987] fmd: change chosen node dt files to support increased
 resources

[Original patch taken from QorIQ-SDK-V1.6-SOURCE-20140619-yocto.iso]

These dt files re-allocates the fman resources to the ports
so that they are stretched close to the fman possible limit

Change-Id: Ia803628186132db8e2969d8c252d869d2cbdb3a8
Signed-off-by: Mandy Lavi <mandy.lavi@freescale.com>
Reviewed-on: http://git.am.freescale.net:8181/13298
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Nir Erez <nir.erez@freescale.com>
Reviewed-by: Thomas Trefny <Tom.Trefny@freescale.com>
---
 .../dts/fsl/qoriq-fman-0-chosen-fifo-resize.dtsi   | 79 ++++++++++++++++++++--
 .../dts/fsl/qoriq-fman-1-chosen-fifo-resize.dtsi   | 79 ++++++++++++++++++++--
 2 files changed, 144 insertions(+), 14 deletions(-)

diff --git a/arch/powerpc/boot/dts/fsl/qoriq-fman-0-chosen-fifo-resize.dtsi b/arch/powerpc/boot/dts/fsl/qoriq-fman-0-chosen-fifo-resize.dtsi
index 17c08d2..fdb78e9 100644
--- a/arch/powerpc/boot/dts/fsl/qoriq-fman-0-chosen-fifo-resize.dtsi
+++ b/arch/powerpc/boot/dts/fsl/qoriq-fman-0-chosen-fifo-resize.dtsi
@@ -38,33 +38,98 @@ chosen {
 		fman0-extd-args {
 			cell-index = <0>;
 			compatible = "fsl,fman-extended-args";
+			total-fifo-size = <0x27000>;
+			fman0_oh1-extd-args {
+				cell-index = <1>;
+				compatible = "fsl,fman-port-op-extended-args";
+				fifo-size = <0x800 0x0>;
+				num-dmas = <0x1 0x1>;
+				num-tnums = <0x1 0x4>;
+			};
 			fman0_rx0-extd-args {
 				cell-index = <0>;
 				compatible = "fsl,fman-port-1g-rx-extended-args";
-				fifo-size = <0x2f00 0>;
+				fifo-size = <0x2f00 0x2000>;
+				num-dmas = <0x1 0x1>;
+				num-tnums = <0x4 0x4>;
+			};
+			fman0_tx0-extd-args {
+				cell-index = <0>;
+				compatible = "fsl,fman-port-1g-tx-extended-args";
+				fifo-size = <0x2c00 0x0>;
+				num-dmas = <0x1 0x0>;
+				num-tnums = <0x4 0x4>;
 			};
 			fman0_rx1-extd-args {
 				cell-index = <1>;
 				compatible = "fsl,fman-port-1g-rx-extended-args";
-				fifo-size = <0x2f00 0>;
+				fifo-size = <0x2f00 0x2000>;
+				num-dmas = <0x1 0x1>;
+				num-tnums = <0x4 0x4>;
+			};
+			fman0_tx1-extd-args {
+				cell-index = <1>;
+				compatible = "fsl,fman-port-1g-tx-extended-args";
+				fifo-size = <0x2c00 0x0>;
+				num-dmas = <0x1 0x0>;
+				num-tnums = <0x4 0x4>;
 			};
 			fman0_rx2-extd-args {
 				cell-index = <2>;
 				compatible = "fsl,fman-port-1g-rx-extended-args";
-				fifo-size = <0x2f00 0>;
+				fifo-size = <0x2f00 0x2000>;
+				num-dmas = <0x1 0x1>;
+				num-tnums = <0x4 0x4>;
+			};
+			fman0_tx2-extd-args {
+				cell-index = <2>;
+				compatible = "fsl,fman-port-1g-tx-extended-args";
+				fifo-size = <0x2c00 0x0>;
+				num-dmas = <0x1 0x0>;
+				num-tnums = <0x4 0x4>;
 			};
 			fman0_rx3-extd-args {
 				cell-index = <3>;
 				compatible = "fsl,fman-port-1g-rx-extended-args";
-				fifo-size = <0x2f00 0>;
+				fifo-size = <0x2f00 0x2000>;
+				num-dmas = <0x1 0x1>;
+				num-tnums = <0x4 0x4>;
+			};
+			fman0_tx3-extd-args {
+				cell-index = <3>;
+				compatible = "fsl,fman-port-1g-tx-extended-args";
+				fifo-size = <0x2c00 0x0>;
+				num-dmas = <0x1 0x0>;
+				num-tnums = <0x4 0x4>;
 			};
 			fman0_rx4-extd-args {
 				cell-index = <4>;
 				compatible = "fsl,fman-port-1g-rx-extended-args";
-				fifo-size = <0x2f00 0>;
+				fifo-size = <0x2f00 0x2000>;
+				num-dmas = <0x1 0x1>;
+				num-tnums = <0x4 0x4>;
+			};
+			fman0_tx4-extd-args {
+				cell-index = <4>;
+				compatible = "fsl,fman-port-1g-tx-extended-args";
+				fifo-size = <0x2c00 0x0>;
+				num-dmas = <0x1 0x0>;
+				num-tnums = <0x4 0x4>;
+			};
+			fman0_rx8-extd-args {
+				cell-index = <0>;
+				compatible = "fsl,fman-port-10g-rx-extended-args";
+				fifo-size = <0x7000 0x3000>;
+				num-dmas = <0x8 0x8>;
+				num-tnums = <0x10 0x8>;
+			};
+			fman0_tx8-extd-args {
+				cell-index = <0>;
+				compatible = "fsl,fman-port-10g-tx-extended-args";
+				fifo-size = <0x4000 0x0>;
+				num-dmas = <0x8 0x0>;
+				num-tnums = <0x10 0x8>;
 			};
 		};
 	};
 };
-
-
diff --git a/arch/powerpc/boot/dts/fsl/qoriq-fman-1-chosen-fifo-resize.dtsi b/arch/powerpc/boot/dts/fsl/qoriq-fman-1-chosen-fifo-resize.dtsi
index d923e55..a555920 100644
--- a/arch/powerpc/boot/dts/fsl/qoriq-fman-1-chosen-fifo-resize.dtsi
+++ b/arch/powerpc/boot/dts/fsl/qoriq-fman-1-chosen-fifo-resize.dtsi
@@ -38,33 +38,98 @@ chosen {
 		fman1-extd-args {
 			cell-index = <1>;
 			compatible = "fsl,fman-extended-args";
+			total-fifo-size = <0x27000>;
+			fman1_oh1-extd-args {
+				cell-index = <1>;
+				compatible = "fsl,fman-port-op-extended-args";
+				fifo-size = <0x800 0x0>;
+				num-dmas = <0x1 0x1>;
+				num-tnums = <0x1 0x4>;
+			};
 			fman1_rx0-extd-args {
 				cell-index = <0>;
 				compatible = "fsl,fman-port-1g-rx-extended-args";
-				fifo-size = <0x2f00 0>;
+				fifo-size = <0x2f00 0x2000>;
+				num-dmas = <0x1 0x1>;
+				num-tnums = <0x4 0x4>;
+			};
+			fman1_tx0-extd-args {
+				cell-index = <0>;
+				compatible = "fsl,fman-port-1g-tx-extended-args";
+				fifo-size = <0x2c00 0x0>;
+				num-dmas = <0x1 0x0>;
+				num-tnums = <0x4 0x4>;
 			};
 			fman1_rx1-extd-args {
 				cell-index = <1>;
 				compatible = "fsl,fman-port-1g-rx-extended-args";
-				fifo-size = <0x2f00 0>;
+				fifo-size = <0x2f00 0x2000>;
+				num-dmas = <0x1 0x1>;
+				num-tnums = <0x4 0x4>;
+			};
+			fman1_tx1-extd-args {
+				cell-index = <1>;
+				compatible = "fsl,fman-port-1g-tx-extended-args";
+				fifo-size = <0x2c00 0x0>;
+				num-dmas = <0x1 0x0>;
+				num-tnums = <0x4 0x4>;
 			};
 			fman1_rx2-extd-args {
 				cell-index = <2>;
 				compatible = "fsl,fman-port-1g-rx-extended-args";
-				fifo-size = <0x2f00 0>;
+				fifo-size = <0x2f00 0x2000>;
+				num-dmas = <0x1 0x1>;
+				num-tnums = <0x4 0x4>;
+			};
+			fman1_tx2-extd-args {
+				cell-index = <2>;
+				compatible = "fsl,fman-port-1g-tx-extended-args";
+				fifo-size = <0x2c00 0x0>;
+				num-dmas = <0x1 0x0>;
+				num-tnums = <0x4 0x4>;
 			};
 			fman1_rx3-extd-args {
 				cell-index = <3>;
 				compatible = "fsl,fman-port-1g-rx-extended-args";
-				fifo-size = <0x2f00 0>;
+				fifo-size = <0x2f00 0x2000>;
+				num-dmas = <0x1 0x1>;
+				num-tnums = <0x4 0x4>;
+			};
+			fman1_tx3-extd-args {
+				cell-index = <3>;
+				compatible = "fsl,fman-port-1g-tx-extended-args";
+				fifo-size = <0x2c00 0x0>;
+				num-dmas = <0x1 0x0>;
+				num-tnums = <0x4 0x4>;
 			};
 			fman1_rx4-extd-args {
 				cell-index = <4>;
 				compatible = "fsl,fman-port-1g-rx-extended-args";
-				fifo-size = <0x2f00 0>;
+				fifo-size = <0x2f00 0x2000>;
+				num-dmas = <0x1 0x1>;
+				num-tnums = <0x4 0x4>;
+			};
+			fman1_tx4-extd-args {
+				cell-index = <4>;
+				compatible = "fsl,fman-port-1g-tx-extended-args";
+				fifo-size = <0x2c00 0x0>;
+				num-dmas = <0x1 0x0>;
+				num-tnums = <0x4 0x4>;
+			};
+			fman1_rx8-extd-args {
+				cell-index = <0>;
+				compatible = "fsl,fman-port-10g-rx-extended-args";
+				fifo-size = <0x7000 0x3000>;
+				num-dmas = <0x8 0x8>;
+				num-tnums = <0x10 0x8>;
+			};
+			fman1_tx8-extd-args {
+				cell-index = <0>;
+				compatible = "fsl,fman-port-10g-tx-extended-args";
+				fifo-size = <0x4000 0x0>;
+				num-dmas = <0x8 0x0>;
+				num-tnums = <0x10 0x8>;
 			};
 		};
 	};
 };
-
-
-- 
1.9.1

